### PR TITLE
enforce strict mypy

### LINKED ISSUE #464: [BUG] catch make style errors in github actions
### 🐛 Bug

Currently, `make style` will report errors that do not yield to failure of automated tests.

### To Reproduce

Run `make style` on latest main

```
llm_studio/src/datasets/text_causal_language_modeling_ds.py:284: error: Argument 1 to "argmax" has incompatible type "ndarray[Any, dtype[Any]]"; expected "Tensor"  [arg-type]
llm_studio/src/utils/modeling_utils.py:101: error: "Tensor" not callable  [operator]
llm_studio/src/utils/modeling_utils.py:521: error: Item "Tensor" of "Tensor | Module" has no attribute "generate"  [union-attr]
llm_studio/src/utils/modeling_utils.py:521: error: "Tensor" not callable  [operator]
train.py:299: error: "Tensor" not callable  [operator]
llm_studio/python_configs/text_causal_language_modeling_config.py:347: error: Incompatible types in assignment (expression has type "float", variable has type "int")  [assignment]
llm_studio/python_configs/text_causal_language_modeling_config.py:348: error: Incompatible types in assignment (expression has type "float", variable has type "int")  [assignment]
llm_studio/python_configs/text_causal_language_modeling_config.py:349: error: Incompatible types in assignment (expression has type "float", variable has type "int")  [assignment]
Found 8 errors in 4 files (checked 75 source files)

```

### LLM Studio version
#[d107d0d](https://github.com/h2oai/h2o-llmstudio/commit/d107d0d7def51f8f9f810312f3444d1bbdece665)

### PR DESCRIPTION
**Pull request by maxjeblick**
Enforce mypy checks. Fixes #464 

Note: For testing, this PR should fail initially as `# type: ignore` comment was disabled in one place.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -53,6 +53,7 @@ style: reports pipenv
 
 	-$(PIPENV) run mypy . --check-untyped-defs | tee -a reports/mypy.log || echo "mypy failed" >> reports/mypy_errors.log
 	@if [ -s reports/mypy_errors.log ]; then exit 1; fi
```
Comment (pascal-pfeiffer) [inline Makefile:55]: let's remove this then

File: Makefile [hunk 2]
```diff
@@ -53,6 +53,7 @@ style: reports pipenv
 
 	-$(PIPENV) run mypy . --check-untyped-defs | tee -a reports/mypy.log || echo "mypy failed" >> reports/mypy_errors.log
```
Comment (pascal-pfeiffer) [inline Makefile:54]: `|| echo "mypy failed" >> reports/mypy_errors.log` is obsolete then

**Review by pascal-pfeiffer — CHANGES_REQUESTED** [review]
small changes, otherwise, lgtm

File: Makefile (hunk 2 above)
Comment (maxjeblick) [reply Makefile]: Thanks, makes sense!

**Review by pascal-pfeiffer — APPROVED** [review]
lgtm, thanks!

**Pull request merged**